tests/installation/npmTest.ts

```diff
@@ -77,25 +63,88 @@ _expect.extend({
-export type ArgsOrOptions = [] | [...string[]] | [...string[], ExecOptions] | [ExecOptions];
+type ExecOptions = { cwd?: string, env?: Record<string, string>, message?: string, expectToExitWithError?: boolean };
+type ArgsOrOptions = [] | [...string[]] | [...string[], ExecOptions] | [ExecOptions];
+
+type NPMTestOptions = {
+  useRealCDN: boolean;
+};
 
-export type NPMTestFixtures = {
-  _auto: void,
-  _browsersPath: string
-  tmpWorkspace: string,
-  nodeMajorVersion: number,
+type NPMTestFixtures = {
+  _auto: void;
+  _browsersPath: string;
+  _cdnProxyEnv: NodeJS.ProcessEnv;
+  tmpWorkspace: string;
   installedSoftwareOnDisk: (registryPath?: string) => Promise<string[]>;
-  writeConfig: (allowGlobal: boolean) => Promise<void>,
-  writeFiles: (nameToContents: Record<string, string>) => Promise<void>,
-  exec: (cmd: string, ...argsAndOrOptions: ArgsOrOptions) => Promise<string>
-  tsc: (...argsAndOrOptions: ArgsOrOptions) => Promise<string>,
-  registry: Registry,
+  writeConfig: (allowGlobal: boolean) => Promise<void>;
+  writeFiles: (nameToContents: Record<string, string>) => Promise<void>;
+  exec: (cmd: string, ...argsAndOrOptions: ArgsOrOptions) => Promise<string>;
+  tsc: (...argsAndOrOptions: ArgsOrOptions) => Promise<string>;
+  registry: Registry;
 };
 
 export const test = _test
     .extend<CommonFixtures, CommonWorkerFixtures>(commonFixtures)
-    .extend<NPMTestFixtures>({
+    .extend<NPMTestFixtures & NPMTestOptions>({
+      useRealCDN: [false, { option: true }],
+      _cdnProxyEnv: async ({ useRealCDN }, use) => {
+        if (useRealCDN) {
+          await use({});
+          return;
+        }
+
+        const server = createHttpServer(async (request: http.IncomingMessage, response: http.ServerResponse) => {
+          const requestedPath = url.parse(request.url!).path;
+          const cachedPath = path.join(BROWSERS_CACHE_DIR, calculateSha1(requestedPath));
+          const cachedPathMetaInfo = cachedPath + '.metainfo';
+
+          if (!fs.existsSync(cachedPath)) {
+            const realUrl = 'https://playwright.azureedge.net' + requestedPath;
+            debug(`[cdn proxy] downloading ${realUrl} headers=${JSON.stringify(request.headers)}`);
+            const headers = { ...request.headers };
+            delete headers['host'];
+            const options = {
+              ...url.parse(realUrl),
+              method: request.method,
+              headers,
+            };
+            const factory = options.protocol === 'https:' ? https : http;
+            let doneCallback = () => {};
+            const donePromise = new Promise<void>(f => doneCallback = () => {
+              debug(`[cdn proxy] downloading ${realUrl} finished`);
+              f();
+            });
+            const realRequest = factory.request(options, (realResponse: http.IncomingMessage) => {
+              const metaInfo = {
+                statusCode: realResponse.statusCode,
```

Comment:
I would appreciate if the `metaInfo` also has a version. This way, if there are changes to this storage format, then no developers would need to manually delete this cache.

Reply:
The cache is actually removed in global setup, so we will not have to worry about this.